File: webui/src/types/preset.ts
export class Preset {
  id: number
  name: string
  description: string | null
  content: { key: string; value: string }[]
  settings: object | null
  isPublic: boolean

  constructor(
    id: number,
    name: string,
    description: string | null = null,
    content: { key: string; value: string }[] = [],
    settings: object | null,
    isPublic: boolean = false,
  ) {
    this.id = id
    this.name = name
    this.description = description
    this.content = content
    this.settings = settings
    this.isPublic = isPublic
  }
}

File: src/Models/Preset.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace LLMRolePlay.Models
{
  public partial class DBContext : Microsoft.EntityFrameworkCore.DbContext
  {
    public DbSet<Preset> Presets { get; set; }
  }

  public class Preset
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]//自增id
    public uint Id { get; set; }
    public required string Name { get; set; }
    public string? Settings { get; set; }
    public required string Content { get; set; }
    public string? Description { get; set; }
    public uint UserId { get; set; }
    public bool IsPublic { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public void MarkAsModified(DBContext db)
    {
      db.Entry(this).State = EntityState.Modified;
    }
  }
}